src/openssl_enc_fmt_plug.c

```diff
@@ -265,27 +272,57 @@ static void *get_salt(char *ciphertext)
 		if (cs.kpa) {
 			p = strtokm(NULL, "$");
 			strncpy((char*)cs.kpt, p, 255);
+			if (cs.kpa == 2) {
+				cs.asciipct = atoi(p);
+				if (!cs.asciipct) {
+					cs.asciipct = 90;
+				}
+			}
 		}
 	}
 
 	MEM_FREE(keeptr);
 	return (void *)&cs;
 }
 
+static int countascii(unsigned char *data, int len) {
+	int nascii = 0;
+	int c;
+	for(c=0; c<len; c++) {
+		if (data[c]==0x0a || (data[c]>=32 && data[c]<127))
+			nascii++;
+	}
+	return nascii;
+}
```

Comment:
You should probably count `0x0d` as well, in case the data has Windows' CRLF line endings.

While at it, please add some air around operators, as in `for (c = 0; c < len; c++)`. That's our code style although I can't remember where it's documented, if at all.

Reply:
Found it... I think it's ok now

Reply:
Almost. I added further minor tweaks in a commit using GitHub's online editor. That was cool, although somewhat scary.